### PR TITLE
GEODE-4312: Wait for rebalancing to complete as part of teardown

### PR DESCRIPTION
**Pull request by jdeppe-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by jinmeiliao** [comment]
thanks for fixing this. I had this failure on my precheckin

File: geode-core/src/test/java/org/apache/geode/test/dunit/cache/internal/JUnit4CacheTestCase.java [hunk 1]
```diff
@@ -368,6 +368,7 @@ public void postTearDownCacheTestCase() throws Exception {
   private static synchronized void remoteTearDown() {
     try {
       DistributionMessageObserver.setInstance(null);
+      waitForNoRebalancing();
```
Comment (agingade) [inline geode-core/src/test/java/org/apache/geode/test/dunit/cache/internal/JUnit4CacheTestCase.java]: Looks like this is test specific...how about having this in "CacheXmlTestCase" instead here...

File: geode-core/src/test/java/org/apache/geode/test/dunit/cache/internal/JUnit4CacheTestCase.java [hunk 2]
```diff
@@ -71,7 +71,7 @@
    * <p>
    * Field is static so it doesn't get serialized with SerializableRunnable inner classes.
    */
-  private static InternalCache cache;
+  protected static InternalCache cache;
```
Comment (pdxrunner) [inline geode-core/src/test/java/org/apache/geode/test/dunit/cache/internal/JUnit4CacheTestCase.java]: Why has the access been changed? There aren't any direct references to `cache` in this change set.

Comment (jdeppe-pivotal) [reply geode-core/src/test/java/org/apache/geode/test/dunit/cache/internal/JUnit4CacheTestCase.java]: The change I made is in a subclass (`CacheXmlTestCase`) and references `cache` there.

Comment (pdxrunner) [reply geode-core/src/test/java/org/apache/geode/test/dunit/cache/internal/JUnit4CacheTestCase.java]: OK, I missed that reference to the cache  during my review.

**Pull request merged**